tests/test_sequence.py

```diff
@@ -137,6 +137,280 @@ def test_magnetic_field():
+        RampWaveform(250, 0.0, 2.3 * 2 * np.pi), -4 * np.pi, 0.0, 0.5
+    )
+    sweep = Pulse.ConstantAmplitude(
+        2.3 * 2 * np.pi, RampWaveform(400, -4 * np.pi, 4 * np.pi), 1.0, 0.6
+    )
+    fall = Pulse.ConstantDetuning(
+        RampWaveform(500, 2.3 * 2 * np.pi, 0.0), 4 * np.pi, 0.0, 0.7
+    )
+    seq.declare_channel("ising", channel_id="rydberg_global")
+    seq.add(rise, "ising")
+    seq.add(sweep, "ising")
+    seq.add(fall, "ising")
+
+    assert seq.switch_device(test_device1)._device == test_device1
+
+    with pytest.raises(
+        ValueError,
+        match="No channel match for channel ising"
+        + " with the right phase_jump_time & clock_period.",
+    ):
+        seq.switch_device(MockDevice, True)
+
+    # Jump_phase_time check 2: No post_phase_shift and pulses with same phase
+    rise = Pulse.ConstantDetuning(
+        RampWaveform(250, 0.0, 2.3 * 2 * np.pi), -4 * np.pi, 0.0
+    )
+    sweep = Pulse.ConstantAmplitude(
+        2.3 * 2 * np.pi, RampWaveform(400, -4 * np.pi, 4 * np.pi), 0.0
+    )
+    fall = Pulse.ConstantDetuning(
+        RampWaveform(500, 2.3 * 2 * np.pi, 0.0), 4 * np.pi, 0.0
+    )
+    seq = Sequence(reg, test_device3)
+    seq.declare_channel(name="ising", channel_id="rydberg_global")
+    seq.add(rise, "ising")
+    seq.add(sweep, "ising")
+    seq.add(fall, "ising")
+
+    with pytest.warns(
+        UserWarning,
+        match="The phase_jump_time of the matching channel"
+        + "on the the new device is different, take it in account"
+        + " for the upcoming pulses.",
+    ):
+        seq.switch_device(Chadoq2, True)
+
+    # Clock_period not match
+    test_device1.change_rydberg_level(test_device2.rydberg_level)
+    seq = Sequence(reg, test_device1)
+    seq.declare_channel("ising", "rydberg_global")
+
+    with pytest.raises(
+        ValueError,
+        match="No channel match for channel ising"
+        + " with the right phase_jump_time & clock_period.",
+    ):
+        seq.switch_device(test_device2, True)
+
+    test_device2.change_rydberg_level(test_device3.rydberg_level)
+    seq = Sequence(reg, test_device3)
+    seq.declare_channel("ising", "rmn_local2", "q0")
+    assert seq.switch_device(test_device2, True)._device == test_device2
+
+    test_device2.change_rydberg_level(test_device1.rydberg_level)
+    seq = Sequence(reg, test_device2)
+    seq.declare_channel("ising", "rmn_local", "q0")
+    with pytest.raises(
+        ValueError,
+        match="No channel match for channel ising"
+        + " with the right mod_bandwidth.",
+    ):
+        seq.switch_device(test_device1, True)
+    test_device3.change_rydberg_level(test_device2.rydberg_level)
+    seq = Sequence(reg, test_device3)
+    seq.declare_channel("ising", "rmn_local1", "q0")
+    with pytest.raises(
+        ValueError,
+        match="No channel match for channel ising"
+        + " with the right fixed_retarget_t.",
+    ):
+        seq.switch_device(test_device2, True)
+
```

Comment:
You should make a successful strict switch on a non-empty sequence and check that the sequence samples all stay the same.